base/server/share/conf/schema.ldif

```diff
@@ -687,3 +687,87 @@ attributeTypes: ( authorityKeyHost-oid NAME 'authorityKeyHost' DESC 'Authority K
+  SYNTAX 1.3.6.1.4.1.1466.115.121.1.24
+  EQUALITY generalizedTimeMatch
+  ORDERING generalizedTimeOrderingMatch
+  SINGLE-VALUE )
+attributeTypes: ( acmeValidatedAt-oid NAME 'acmeValidatedAt'
+  SYNTAX 1.3.6.1.4.1.1466.115.121.1.24
+  EQUALITY generalizedTimeMatch
+  ORDERING generalizedTimeOrderingMatch
+  SINGLE-VALUE )
+attributeTypes: ( acmeStatus-oid NAME 'acmeStatus'
+  SYNTAX 1.3.6.1.4.1.1466.115.121.1.15
+  EQUALITY caseIgnoreMatch
+  SINGLE-VALUE )
+attributeTypes: ( acmeError-oid NAME 'acmeError'
+  SYNTAX 1.3.6.1.4.1.1466.115.121.1.15
+  SINGLE-VALUE )
+attributeTypes: ( acmeNonceValue-oid NAME 'acmeNonceValue'
+  SUP name
+  SINGLE-VALUE )
+attributeTypes: ( acmeAccountId-oid NAME 'acmeAccountId'
+  SUP name
+  SINGLE-VALUE )
+attributeTypes: ( acmeAccountContact-oid NAME 'acmeAccountContact'
+  SYNTAX 1.3.6.1.4.1.1466.115.121.1.15
+  EQUALITY caseIgnoreMatch
+  SUBSTR caseIgnoreSubstringsMatch )
+attributeTypes: ( acmeAccountKey-oid NAME 'acmeAccountKey'
+  SYNTAX 1.3.6.1.4.1.1466.115.121.1.15
+  SINGLE-VALUE )
+attributeTypes: ( acmeOrderId-oid NAME 'acmeOrderId'
+  SUP name
+  SINGLE-VALUE )
+attributeTypes: ( acmeIdentifier-oid NAME 'acmeIdentifier'
+  SYNTAX 1.3.6.1.4.1.1466.115.121.1.15
+  EQUALITY caseIgnoreMatch )
+attributeTypes: ( acmeAuthorizationId-oid NAME 'acmeAuthorizationId'
+  SUP name )
+attributeTypes: ( acmeAuthorizationWildcard-oid NAME 'acmeAuthorizationWildcard'
+  SYNTAX 1.3.6.1.4.1.1466.115.121.1.7
+  EQUALITY booleanMatch
+  SINGLE-VALUE )
+attributeTypes: ( acmeChallengeId-oid NAME 'acmeChallengeId'
+  SUP name
+  SINGLE-VALUE )
+attributeTypes: ( acmeToken-oid NAME 'acmeToken'
+  SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 )
+attributeTypes: ( acmeCertificateId-oid NAME 'acmeCertificateId'
+  SYNTAX 1.3.6.1.4.1.1466.115.121.1.15
+  EQUALITY caseExactMatch
+  SINGLE-VALUE )
+-
+add: objectClasses
+objectClasses: ( acmeNonce-oid NAME 'acmeNonce'
+  STRUCTURAL
+  MUST ( acmeNonceValue $ acmeExpires ) )
+objectClasses: ( acmeAccount-oid NAME 'acmeAccount'
+  STRUCTURAL
+  MUST ( acmeAccountId $ acmeAccountKey $ acmeStatus )
+  MAY acmeAccountContact )
+objectClasses: ( acmeOrder-oid NAME 'acmeOrder'
+  STRUCTURAL
+  MUST ( acmeOrderId $ acmeAccountId $ acmeStatus $ acmeIdentifier $ acmeAuthorizationId )
+  MAY ( acmeError $ acmeCertificateId $ acmeExpires ) )
+objectClasses: ( acmeAuthorization-oid NAME 'acmeAuthorization'
+  STRUCTURAL
+  MUST ( acmeAuthorizationId $ acmeAccountId $ acmeIdentifier $ acmeStatus )
+  MAY ( acmeExpires $ acmeAuthorizationWildcard ) )
+objectClasses: ( acmeChallenge-oid NAME 'acmeChallenge'
+  ABSTRACT
+  MUST ( acmeChallengeId $ acmeAccountId $ acmeAuthorizationId $ acmeStatus )
+  MAY ( acmeValidatedAt $ acmeError ) )
+objectClasses: ( acmeChallengeDns01-oid NAME 'acmeChallengeDns01'
+  SUP acmeChallenge
+  STRUCTURAL
+  MUST acmeToken )
+objectClasses: ( acmeChallengeHttp01-oid NAME 'acmeChallengeHttp01'
+  SUP acmeChallenge
+  STRUCTURAL
+  MUST acmeToken )
```

Comment:
These object classes are identical. Would it be better to just use `acmeChallenge` and store the type in an attribute. That way if we support another challenge type we don't have to change the schema.

Reply:
The motivation is that future challenge types could involve other data that need to be stored, so we would want a different objectclass for it, to add the new fields.  I'll make sure to explain this in the design doc.

Reply:
OK.